### DISCUSSION TITLE
Python Api import issue: pip and ssl for 'ffpython.exe' are broken!

### DISCUSSION THREAD

**Discussion by NultiluK** [fontforge/fontforge#5258] [Q&A]
env: Windows 11, Pycharm2023.1 Python3.11
My work is to convert some pngs into a .ttf file. When I use fonforge's python api, some problems I can't figure out.
![image](https://github.com/fontforge/fontforge/assets/48043086/7ed8c802-bd8d-4e48-adea-a36c75b1232e)
As u see in figure above, I configed a normal interpreter and a fontforge interpreter, which is using ffpython.exe. In the normal interpreter, I import some packges, such as pillow. But in this env, `import fontforge` will fail. So i also configed a fontforge to import this package succesfully and it worked. 
And here is the problem in this interpreter, the first problem is pip. When I using 'pip install pillow', the pip in this software's lib is broken, which shows 'pip._vendor.distlib' error. I solved it by coping my global python pip lib to cover the software's pip lib. Then the second problem is unfold. Again I use 'pip install pillow', I found the ssl package is not exist! I googled\ chatgpt every solution, thier solutions always depend on pip, but this pip lacks ssl or something i have not met. I tried to use global enviroment package to create this interpreter, but this also need a good pip with ssl.
How can 'import fontforge' work in the global enviroment? OR How the "fontforge" enviroment can use global python package directly, without use itself's broken pip?

**Comment by ctrlcctrlv** [comment]
The problem is almost certainly mingw (Minimalist GNU¹ for Windows) Python vs. MSVC² Python.

Your system Python is most likely MSVC Python, while FontForge still only compiles under mingw.

This is a complex issue and the SSL package likely does not exist because it wants to compile or link against C code (probably OpenSSL or GNUTLS).

Consider using MSYS2 and sticking to one or the other, MinGW Python or MSVC Python—as stated, for FontForge you need the former.

# Footnotes
- ¹ As in compiler and glibc.
- ² Microsoft Visual C(++)

**Reply by ctrlcctrlv** [reply]
Note: @JayusKang I moved your issue to a discussion as it had been a few days w/o response and I'm quite sure I'm right. If I got it wrong, and a bug truly exists, please open it against [fontforge/fontforgebuilds](https://github.com/fontforge/fontforgebuilds), which tracks Win32 issues.
